pulpcore/cli/common/generic.py

```diff
@@ -344,6 +357,13 @@ def _multi_option_callback(
     expose_value=False,
 )
 
+id_option = pulp_option(
+    "--id",
```

Comment:
This is basically "name-or-href" now. At that point we could almost make it a positional argument. (Do you sense that i am not fond of the term "id" here? Don't think it is very instructive to users.)
But, what it you need two data to identify one entity? i.e. digest, rel-path...

Reply:
That is: `pulp file repository sync <name_or_href>`.

Reply:
Yea, I'm not fond of the option name either but it seems necessarily ambiguous/generic to capture both name and href. It's also short.

I'm a bit mixed about introducing the concept of positional arguments. More UI patterns introduce more mental tax.

I'll think about this feature some more.